mirascope/chat/utils.py

```diff
@@ -12,26 +12,45 @@
 from .tools import OpenAITool, openai_tool_fn
 
 
-def _get_openai_chat_messages(
-    prompt: Prompt,
-) -> list[ChatCompletionMessageParam]:
-    """Returns a list of messages parsed from the prompt."""
-    return [
-        cast(ChatCompletionMessageParam, {"role": role, "content": content})
-        for role, content in prompt.messages
-    ]
-
-
 def get_openai_messages_from_prompt(
     prompt: Union[Prompt, str],
 ) -> list[ChatCompletionMessageParam]:
     """Returns a list of messages parsed from the prompt."""
     if isinstance(prompt, Prompt):
-        return _get_openai_chat_messages(prompt)
+        return [
+            cast(ChatCompletionMessageParam, {"role": role, "content": content})
+            for role, content in prompt.messages
+        ]
     else:
         return [cast(ChatCompletionMessageParam, {"role": "user", "content": prompt})]
 
 
+def setup_openai_kwargs(
+    kwargs: dict[str, Any],
+    prompt: Optional[Union[Prompt, str]],
+    tools: Optional[list[Union[Callable, Type[OpenAITool]]]],
+) -> Optional[list[Type[OpenAITool]]]:
+    """Sets up the kwargs for an OpenAI API call."""
+    if prompt is None:
+        if "messages" not in kwargs:
+            raise ValueError("Either `prompt` or `messages` must be provided.")
+    else:
+        kwargs["messages"] = get_openai_messages_from_prompt(prompt)
+
+    if tools:
+        openai_tools: list[type[OpenAITool]] = [
+            tool if isclass(tool) else convert_function_to_openai_tool(tool)
+            for tool in tools
+        ]
+        kwargs["tools"] = [tool.tool_schema() for tool in openai_tools]
+        if "tool_choice" not in kwargs:
+            kwargs["tool_choice"] = "auto"
+
+        return openai_tools
```

Comment:
we currently arent doing anything with this return so I would omit it completely. that way this function is very clear in what it is doing which is mutating kwargs, whereas currently it feels like its mutating kwargs but returning something slightly different